homeassistant/util/hass_dict.pyi

```diff
@@ -9,8 +9,7 @@ __all__ = [
     "HassKey",
 ]
 
-_T = TypeVar("_T")
-_U = TypeVar("_U")
+_T = TypeVar("_T")  # needs to be invariant
```

Comment:
> [!NOTE]
> We can't use the new syntax here! For PEP 695, type checkers infer the variance base on how the TypeVar is used inside a class. As it isn't used at all, it would be inferred as both covariant and contravariant, essentially making it useless.
>
> The old `TypeVar("_T")` declaration is always invariant.